### DISCUSSION TITLE
Simplest Way to Share Dir with All Clients?

### DISCUSSION THREAD

**Discussion by stevehol** [ltsp/ltsp#282] [Q&A]
I have a directory /home/share/ with files to be made available read-only to all client users. 755 mode allows any user to access this dir if they are logged on at the server directly, but if they are logged on at a client they cannot see anything in /home except their own home tree.

I can create a mount point in each user's home, and a bind mount in etc/fstab to /home/share, but I wonder if there is an even simpler way to allow all users access to /home/share/ without configuring each user?

I guess this would involve a config for SSHFS or exporting an NFS share?

**Comment by alkisg** [comment]
Right, you'd create an NFS export for that in the server /etc/exports.d/local.conf,
and you'd use `FSTAB_HOME_SHARED="...the fstab line"` in ltsp.conf to have it mounted in the clients.

**Comment by stevehol** [comment]
@alkisg thanks for the pointers. I should mention I am a total NFS-newbie!
I found I had to name the file 'local.exports'.
I added the single line:-
`/home/share 192.168.123.123(ro,sync,no_subtree_check)` where IP is that of the client.
Then after `# exportfs -ra`
`# showmount -e` on the server listed the share, and 
`$ showmount -e 192.168.456.456` on the client (IP is the server) listed the share too.

I added the ltsp.conf entry under [clients]:-
`FSTAB_HOME_SHARED="192.168.456.456:/home/share /home/<user>/share nfs ro,hard 0 0"`
and did `# ltsp initrd`

But when I boot the client and login as this user, he is unable to access the content.
'Failed to mount "share" ... operation permitted for root only'
Permissions on `/home/share` and `/home/<user>/share` are same: root:root 755
The same share via a bind mount works fine.

Its as if the mount in the FSTAB_HOME_SHARED entry is not executed as root?
I tried moving the entry to the [common] section but the error is exactly the same.

**Comment by alkisg** [comment]
On the client, run `cat /etc/fstab`.
If the line is there, you did things correctly.
If not, you did something wrong, e.g. you might have forgotten to run `ltsp initrd` to export ltsp.conf.

Also note that there's no IP=192.168.456.456, I guess that was meant to be an example. There's no need to hide internal IPs, they're not really sensitive information as they're not accessible by the Internet.
You can also put "server" instead of the IP there, as LTSP clients know the LTSP server with that name.

**Comment by stevehol** [comment]
`$ cat /etc/fstab` on the client gave only: '# Generated by 'ltsp init' - no mount
But I definitely did ltsp initrd.

Replacing the server IP with the word 'server' gets the mount into /etc/fstab on the client, and its now displayed as expected by `$ cat /etc/fstab`.
The IP was definitely correct - I also had to use the 'SERVER=' in ltsp.conf and the two IPs were the same. Perhaps FSTAB param only works with 'server'.

But I still have the original issue! 
The client user still gets 'Failed to mount "share" ... operation permitted for root only' in thunar, and 
`$ ls ~/share` still shows nothing.

**Comment by alkisg** [comment]
`ls ~/share` would list `/home/username/share`. I thought you wanted `/home/share`.
Does that directory exist? If not, you have to create it with e.g.

```shell
POST_INIT_MD_SHARE="mkdir -p /home/share"
```

You could also send more information, like the output of the `mount` command on the client or of `ls /home` or of `cat /etc/fstab`. But in general, since the FSTAB line did go into the client /etc/fstab, it's no longer an LTSP issue, it's a regular NFS sharing issue, not related to LTSP.

**Comment by stevehol** [comment]
> ls ~/share would list /home/username/share. I thought you wanted /home/share.

I want the client to have access to /home/share, but the only way I could see of doing it was to mount /home/share on /home/username/share?
/home/username/share does exist.

Is it required to use the `FSTAB_HOME=...` param before using `FSTAB_HOME_SHARED` ?

**Comment by alkisg** [comment]
No. If you're using SSHFS home, then /home is NOT mounted, /home/user is mounted on login via SSHFS, and /home/share is mounted via NFS.

If you're using NFS /home, then you do not need anything else, this will share everything under /home anyway.
